specs/paymentrequest.html

```diff
@@ -284,16 +284,78 @@
           payment app.
         </div>
         <div class="note">
-          <p>The <code>supportedMethods</code> sequence contains the <a>payment method identifiers</a>
-          for the <a>payment methods</a> that the merchant web site accepts.</p>
+
+          <p>The <code>details</code> contains the core payment request data that will be passed on to the user selected
+            <a>payment app</a>.</p>
+
+          <p>The <code>amount</code> sequence contains the amount (possibly in different currencies) that the website
+            is requesting to be paid.</p>
+
+          <p>The <code>supportedMethods</code> sequence contains the <a>payment methods</a> that the merchant web
+            site accepts. Each <a>payment method</a> is linked to one or more <a>payment method identifiers</a>.</p>
+
+          <p>The <code>identifers</code> sequence contains the <a>payment method identifiers</a> that identify the
+          <a>payment methods</a> the web site supports and for which the associated <code>data</code> is part of the
+          request.</p>
+
+          <p><code>data</code> is a <a>JSON-serializable object</a> that provides optional information that might
+          be needed by the supported <a>payment methods</a>.</p>
+
           <pre class="example highlight">
-            ["visa", "bitcoin", "bobpay.com"]
+            {
+              "amount" : [{
```

Comment:
@zkoch We have had strong support for including this.

I understand that it makes the UX more complex but that's an implementer challenge that I assumed would be embraced as a way to differentiate.

Off the top of my head I can imagine a browser having a small selector that allows the user to select their payment currency and when they do the rest of the UI adjusts to display the pricing in that currency. The selector would only be visible if the request actually contains more than one currency.

On the other hand, I don't see us having any conformance language that forces the implementer to display all currencies to the user in the app selection UI. If the browser picks one currency and displays that currency and price that is still valid as the user's option to select another currency will then be made in the app.